linked_list/main.py

```diff
@@ -62,34 +66,129 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
+        self.start = Node(elem=None)
+        self.end = self.start
+        self.counter = 0
+        self.i = self.start
+        if elements is not None:
+            for elem in elements:
+                if self.end.elem == None:
+                    self.end.elem = elem
+                else:
+                    self.end.next = Node(elem=elem)
+                    self.end = self.end.next
 
     def __str__(self):
-        pass
+        return str([ x for x in self ])
 
     def __len__(self):
-        pass
+        return self.count()
 
     def __iter__(self):
-        pass
+        return self
+
+    def __next__(self):
+        if self.counter < self.count():
+            elem = self.i.elem
+            self.counter += 1
+            self.i = self.i.next
+            return elem
+        else:
+            self.counter = 0
+            self.i = self.start
+            raise StopIteration
```

Comment:
Usually the "cleaning up" process in an iterator is done in the __iter__() method.